eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -757,11 +758,18 @@ public function moveSubtree(APILocation $location, APILocation $newParentLocatio
     public function deleteLocation(APILocation $location)
     {
         $location = $this->loadLocation($location->id);
+        $contentInfo = $location->contentInfo;
+        $versionInfo = $this->persistenceHandler->contentHandler()->loadVersionInfo(
+            $contentInfo->id,
+            $contentInfo->currentVersionNo
+        );
+        $translations = $versionInfo->languageCodes;
```

Comment:
Please check if you could simplify that like this
```suggestion
        $translations = $location->getContent()->getVersionInfo()->languageCodes;
```

Reply:
If we simplify that then an additional `content/read` policy will be needed to perform this action